karrot/subscriptions/receivers.py

```diff
@@ -212,7 +212,12 @@ def send_group_detail(group, user=None):
 
 def send_group_preview(group):
     for subscription in ChannelSubscription.objects.recent():
-        preview_payload = GroupPreviewSerializer(group, context={'request': MockRequest(user=subscription.user)}).data
+        preview_payload = GroupPreviewSerializer(
+            group,
+            context={
+                'request': MockRequest(user=subscription.user, META={'REMOTE_ADDR': subscription.client_ip})
+            }
+        ).data
```

Comment:
There are other places that use a `MockRequest` in the context of a subscription, maybe worth a `subscription -> MockRequest` function, and using it in the other places too?

Reply:
Yup, good idea! I'll remember this for another PR :smile: